### PR TITLE
Update yaml plugin

### PR DESCRIPTION
**Pull request by jesper-friis**
# Description:
Fixed some bugs in the yaml plugin:
- Don't dublicate UUID in YAML output.
- Update default to not overwrite yaml files (in single-entity form) that contains an entity.

## Type of change:
- [x] Bug fix.
- [ ] New feature.
- [ ] Documentation update.

## Checklist for the reviewer:

This checklist should be used as a help for the reviewer.

- [ ] Is the change limited to one issue?
- [ ] Does this PR close the issue?
- [ ] Is the code easy to read and understand?
- [ ] Do all new feature have an accompanying new test?
- [ ] Has the documentation been updated as necessary?

### CODE REVIEW

File: storages/python/python-storage-plugins/yaml.py [hunk 1]
```diff
@@ -89,7 +93,8 @@ def save(self, inst: dlite.Instance):
 
         """
         self._data[inst.uuid] = inst.asdict(
-            soft7=dlite.asbool(self.options.soft7)
+            soft7=dlite.asbool(self.options.soft7),
+            uuid=self.single,
```
Comment (CasperWA) [inline storages/python/python-storage-plugins/yaml.py:97]: Not sure what this means. But I'm guessing it means that only a single instance from `inst` is returned? Or multiple?

File: storages/python/python-storage-plugins/yaml.py [hunk 2]
```diff
@@ -74,12 +73,17 @@ def load(self, id: str):
...
             self._data,
             id,
             single=self.options.single,
             check_storages=False,
         )
+        # Metadata in single-entity form is always read-only
+        if inst.is_meta and self.single:
+            self.writable = False
+
+        return inst
```
Comment (CasperWA) [inline storages/python/python-storage-plugins/yaml.py:86]: Is this ensuring the instance cannot be overwritten? This confuses me a bit. Here we've already loaded the instance, right? So setting the `writable` property afterwards - does that change anything?

File: storages/python/python-storage-plugins/yaml.py (hunk 1 above)
Comment (jesper-friis) [reply storages/python/python-storage-plugins/yaml.py:97]: It means that it is only when we we to a yaml file in single-entity form that we want to include the UUID. In multiple-entity form, the UUID (or possible the URI) will already be provided as the key for the entity.

File: storages/python/python-storage-plugins/yaml.py (hunk 2 above)
Comment (jesper-friis) [reply storages/python/python-storage-plugins/yaml.py:86]: Yes, it makes a difference. It is not the instance we want to write-protect, but the yaml file. 

It is first after we have loaded the instance we know whether it is a metadata in single-entity form.

File: storages/python/python-storage-plugins/yaml.py [hunk 3]
```diff
@@ -74,12 +73,17 @@ def load(self, id: str):
...
             A DLite Instance corresponding to the given `id` (UUID).
         """
-        return instance_from_dict(
+        inst = instance_from_dict(
             self._data,
             id,
             single=self.options.single,
             check_storages=False,
         )
+        # Metadata in single-entity form is always read-only
```
Comment (francescalb) [inline storages/python/python-storage-plugins/yaml.py:82]:
```suggestion
        # Ensure metadata in single-entity form is always read-only
```

**Review by francescalb — APPROVED** [review]
Approved with one comment.

**Pull request merged**